### PR TITLE
Added Equals to Result  

### PR DESCRIPTION
**Pull request by theonlyroboboro**

### CODE REVIEW

**Comment by theonlyroboboro** [comment]
partial #35

File: src/DeFuncto.Core/Types/Result.cs [hunk 1]
```diff
@@ -114,6 +114,18 @@ public Unit Iter(Action<TOk> iteratorOk, Action<TError> iteratorError)
...
         public static implicit operator Result<TOk, TError>(TError error) => Error(error);
+
+        public override bool Equals(object obj)
+        {
+            var nval = obj as Result<TOk, TError>?;
+            return nval != null && Equals((Result<TOk, TError>)nval);
+        }
+
+        public bool Equals(Result<TOk, TError> other) =>
+            IsOk == other.IsOk && other.value.Equals(value);
```
Comment (JYCabello) [inline src/DeFuncto.Core/Types/Result.cs:125]: The IsOk is implicitly tested in the equals of the `Du`, just test the equality of the DU.

**Review by JYCabello — APPROVED** [review]

File: src/DeFuncto.Core/Types/Result.cs [hunk 2]
```diff
@@ -114,6 +114,18 @@ public Unit Iter(Action<TOk> iteratorOk, Action<TError> iteratorError)
         [Pure]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static implicit operator Result<TOk, TError>(TError error) => Error(error);
+
+        public override bool Equals(object obj)
+        {
+            var nval = obj as Result<TOk, TError>?;
```
Comment (JYCabello) [inline src/DeFuncto.Core/Types/Result.cs:120]: Look at the version with pattern matching in `Du`

File: src/DeFuncto.Tests/Core/Types/Result/GetHashCode.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+﻿using FsCheck;
+using Xunit;
+using static DeFuncto.Prelude;
```
Comment (JYCabello) [inline src/DeFuncto.Tests/Core/Types/Result/GetHashCode.cs:3]: Static imports last.

File: src/DeFuncto.Tests/Core/Types/Result/Equals.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
+﻿using FsCheck;
+using DeFuncto.Extensions;
+using Xunit;
+using static DeFuncto.Prelude;
```
Comment (JYCabello) [inline src/DeFuncto.Tests/Core/Types/Result/Equals.cs:4]: Static imports last.

**Review by JYCabello — CHANGES_REQUESTED** [review]

**Review by JYCabello — APPROVED** [review]

**Pull request merged**